### DISCUSSION TITLE
File & related collection references in API

### DISCUSSION THREAD

**Discussion by eliteproxy7** [directus/directus#5808] [Archive]
When i have an item with an image or file upload field when I call the item from the API, the image value is the hash, so I have to make a second call to get the image URL/name. Is there a way to specify to show the image path/name in the api call instead?

in some way of a related second question can the same thing be done for other related collections? When i add a O2M relation it shows the item id, i would like for it to include all the fields form the related collection nested in the call

**Comment by rijkvanzanten** [comment]
https://docs.directus.io/reference/api/items/#query-parameters
https://docs.directus.io/reference/api/query/#fields

**Reply by eliteproxy7** [reply]
wow! this just saved so much time for me and I feel bad for even asking since it was clearly in the documentation. Thank you!
